lib/Intlc/Backend/TypeScript/Compiler.hs

```diff
@@ -15,9 +15,12 @@ import           Prelude                           hiding (Type)
 import           Utils                             ((<>^))
 
 compileNamedExport :: InterpStrat -> Locale -> Text -> ICU.Message -> Text
-compileNamedExport x l k v =
-  "export const " <> n <> ": " <> compileTypeof x v <> " = " <> arg <> " => " <> r
-  where (n, r) = JS.compileStmtPieces x l k v
+compileNamedExport s l k v = JS.compileStmt o s l k v
+  where o = JS.emptyOverrides { JS.stmtOverride = Just stmt, JS.matchLitCondOverride = Just matchLitCond }
+        stmt n r = "export const " <> n <> ": " <> compileTypeof s v <> " = " <> arg <> " => " <> r
+        -- Prevents TypeScript from narrowing, which absent this causes some
+        -- nested switch output to fail typechecking.
```

Comment:
As far as I understand TS is doing the right thing in normal circumstances (e.g application code) but here since we duplicate code on purpose, it doesn't suit us right?

Reply:
Correct. We could complicate the compiler to remove unneeded nested cases, but the current compiler implementation tries to be as simple as possible.